cmd/podman/images/scan.go

```diff
@@ -0,0 +1,117 @@
+package images
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/containers/podman/v2/cmd/podman/common"
+	"github.com/containers/podman/v2/cmd/podman/registry"
+	"github.com/containers/podman/v2/libpod/define"
+	"github.com/containers/podman/v2/pkg/domain/entities"
+	"github.com/containers/podman/v2/pkg/specgen"
+	"github.com/spf13/cobra"
+)
+
+var (
+	imageScanCommand     = &cobra.Command{
+		Use:     "scan [run-options] IMAGE [scanner-options]",
+		Short:   "Scan a container image for known vulnerabilities",
+		Long:    `Scan a container image for known vulnerabilities. The image name or digest can be used.`,
+		Args: cobra.MinimumNArgs(1),
+		RunE:    scan,
+		Example: `podman image scan centos:latest`,
+	}
+
+	scanOptions entities.ImageScanOptions
+)
+
+
+func scanFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+
+	// TODO: derive default from containers.conf configuration
+	scannerFlagName := "scanner"
+	flags.StringVar(&scanOptions.ScannerImage, scannerFlagName, "docker.io/anchore/grype:latest", "The vulnerability scanner container image to use")
+}
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode},
+		Command: imageScanCommand,
+		Parent:  imageCmd,
+	})
+	scanFlags(imageScanCommand)
+}
+
+func scan(cmd *cobra.Command, args []string) error {
+	imageToScan := args[0]
+	mountTarget := "/podman-scan-image-mount"
+	// allow parsing to continue with the first arg being the scanner image (not the image to scan)
+	args[0] = scanOptions.ScannerImage
+
+	ctrConfig := registry.PodmanConfig()
+
+	ctrOpts := common.ContainerCLIOpts{
+		Mount: []string{fmt.Sprintf("type=image,src=%s,target=%s,readwrite=false", imageToScan, mountTarget)},
```

Comment:
I don't think that any options but `Mount` are needed.  The backend should automatically add defaults in case the fields aren't set.

Reply:
In case I am wrong, please add a comment why these options are needed.